tests/test-cmd/test.run.js

```diff
@@ -28,48 +58,279 @@ describe('run', () => {
+    return cmd.run().then(() => {
+      let install = cmd.options.firefox.installExtension;
+      assert.equal(install.called, true);
+      assert.equal(
+          install.firstCall.args[0].manifestData.applications.gecko.id,
+          'minimal-example@web-ext-test-suite');
+      assert.deepEqual(install.firstCall.args[0].profile, profile);
+      assert.match(install.firstCall.args[0].extensionPath,
+                   /minimal_extension-1\.0\.xpi/);
 
-        assert.equal(fakeFirefox.run.called, true);
-        assert.deepEqual(fakeFirefox.run.firstCall.args[0], profile);
-      });
+      assert.equal(firefox.run.called, true);
+      assert.deepEqual(firefox.run.firstCall.args[0], profile);
+    });
   });
 
   it('passes a custom Firefox binary when specified', () => {
-    let firefoxBinary = '/pretend/path/to/Firefox/firefox-bin';
-    let fakeFirefox = getFakeFirefox();
-    return runMinimalExt({firefoxBinary}, {firefox: fakeFirefox})
-      .then(() => {
-        assert.equal(fakeFirefox.run.called, true);
-        assert.equal(fakeFirefox.run.firstCall.args[1].firefoxBinary,
-                     firefoxBinary);
-      });
+    const firefoxBinary = '/pretend/path/to/Firefox/firefox-bin';
+    const cmd = prepareRun();
+    const {firefox} = cmd.options;
+
+    return cmd.run({firefoxBinary}).then(() => {
+      assert.equal(firefox.run.called, true);
+      assert.equal(firefox.run.firstCall.args[1].firefoxBinary,
+                   firefoxBinary);
+    });
   });
 
   it('passes a custom Firefox profile when specified', () => {
-    let firefoxProfile = '/pretend/path/to/firefox/profile';
-    let fakeFirefox = getFakeFirefox();
-    return runMinimalExt({firefoxProfile}, {firefox: fakeFirefox})
-      .then(() => {
-        assert.equal(fakeFirefox.createProfile.called, false);
-        assert.equal(fakeFirefox.copyProfile.called, true);
-        assert.equal(fakeFirefox.copyProfile.firstCall.args[0],
-                     firefoxProfile);
+    const firefoxProfile = '/pretend/path/to/firefox/profile';
+    const cmd = prepareRun();
+    const {firefox} = cmd.options;
+
+    return cmd.run({firefoxProfile}).then(() => {
+      assert.equal(firefox.createProfile.called, false);
+      assert.equal(firefox.copyProfile.called, true);
+      assert.equal(firefox.copyProfile.firstCall.args[0],
+                   firefoxProfile);
+    });
+  });
+
+  it('can watch and reload the extension', () => {
+    const cmd = prepareRun();
+    const {sourceDir, artifactsDir} = cmd.argv;
+    const {reloadStrategy} = cmd.options;
+
+    return cmd.run({noReload: false}).then(() => {
+      assert.equal(reloadStrategy.called, true);
+      const args = reloadStrategy.firstCall.args[0];
+      assert.equal(args.sourceDir, sourceDir);
+      assert.equal(args.artifactsDir, artifactsDir);
+      assert.typeOf(args.createRunner, 'function');
+    });
+  });
+
+  it('allows you to opt out of extension reloading', () => {
```

Comment:
how about adding another test case which asserts the expected default behavior

e.g.:
```it('reload the extension on source changes by default', ...)``` or ```it('does not reload the extension on source changes by default')``` (which should be probably tested in the `test.program.js` file where we can test that the options is defined in `program.js` with the expected default value, because we already test the watcher or the reload strategy elsewhere)